### ISSUE TITLE
Under the premise of setting the random seed, the same configuration was run three times and the test results differed by 2%.

### ISSUE THREAD

**Issue by lijiaoyang** [TIGER-AI-Lab/VLM2Vec#72]
@XMHZZ2018 Hello author, I tried an experiment on a small-scale dataset. First, I set the random seed in train.py. The training script is as follows. I ran it 3 times and then performed eval.py evaluation on 36 evaluation sets. The results differed by 2%. **Have you encountered this phenomenon?**

1. set the random seed in train.py at line 18: https://github.com/TIGER-AI-Lab/VLM2Vec/blob/main/train.py#L18
```
from transformers import set_seed
set_seed(42)
```

2. training script
```
torchrun --nproc_per_node=8 --master_port=22447 --max_restarts=0 train.py \
  --model_name /xxx/Qwen2-VL-2B-Instruct/ \
  --output_dir ./mmeb-qwen-test-base5k-run1 \  # e.g. run2,run3
  --bf16 --pooling last \
  --lora --lora_r 8 \
  --dataset_name /xxx/dataset/MMEB-train \
  --split_name original \
  --subset_name VisualNews_i2t VisDial ImageNet_1K OK-VQA ChartQA NIGHTS A-OKVQA CIRR Visual7W InfographicsVQA SUN397 DocVQA MSCOCO_i2t HatefulMemes N24News VOC2007 MSCOCO_t2i WebQA MSCOCO VisualNews_t2i \
  --num_sample_per_subset 5000 \
  --image_dir /xxx/dataset/MMEB-train \
  --image_resolution high --max_len 4096 \
  --logging_steps 1 \
  --lr_scheduler_type linear --learning_rate 2e-5 --num_train_epochs 1 \
  --warmup_steps 200 --save_steps 1000000000 --normalize True \
  --temperature 0.02 --per_device_train_batch_size 16 \
  --grad_cache True --gc_q_chunk_size 4 --gc_p_chunk_size 4 \
  --save_safetensors False --remove_unused_columns False \
  --ddp_find_unused_parameters False \
  --report_to 'tensorboard'
```

3. eval script
```
dataname='MSCOCO VisualNews_t2i Place365 ImageNet-A ImageNet-R ObjectNet Country211 ScienceQA VizWiz GQA TextVQA OVEN FashionIQ EDIS Wiki-SS-NQ Visual7W-Pointing RefCOCO RefCOCO-Matching VisualNews_i2t VisDial ImageNet-1K OK-VQA ChartQA NIGHTS A-OKVQA CIRR Visual7W InfographicsVQA SUN397 DocVQA MSCOCO_i2t HatefulMemes N24News VOC2007 MSCOCO_t2i WebQA'
python eval.py --lora --model_name /xxx/Qwen2-VL-2B-Instruct/ \
  --checkpoint_path ./mmeb-qwen-test-base5k-run1/ \
  --model_backbone qwen2_vl \
  --encode_output_path eval_outputs/qwenvl2B_5k_epoch1_run1/ \
  --image_resolution high --max_len 4096 \
  --pooling last --normalize True \
  --dataset_name /xxx/dataset/MMEB-eval \
  --subset_name ${dataname} \
  --dataset_split test --per_device_eval_batch_size 32 \
  --image_dir /xxx/dataset/MMEB-eval \
  --dataloader_num_workers 4

set +x

```
4. The following is a comparison of evaluation results.

![Image](https://github.com/user-attachments/assets/eac46c52-41a4-44e6-9c39-c976cf5f5974)

5. The training loss curves are almost the same

![Image](https://github.com/user-attachments/assets/4d4cdd28-7198-4a68-99e1-4d4b8283644a)

**Comment by TianhangXiang** [comment]
I also encountered the problem. After fixing all the seeds, I ran the model with the same data and noticed the loss and the grad_norm difference after a few training steps. I tried turning off the flash attention, and then the loss became consistent across different runs. It may be caused by the flash-attention-2.
